dronegen/container_images.go

```diff
@@ -0,0 +1,863 @@
+// sed -i '' 's~/go/~/drone/src/go/~g' .drone.yml
+// # Pull the current branch instead of v10
+// sed -i '' "s~git checkout -qf \"\$(cat '/go/vars/full-version/v10')\"~git checkout -qf \"${DRONE_SOURCE_BRANCH}\"~" .drone.yml
+// # `drone exec` does not properly map the workspace path. This creates a volume to be shared between steps
+// #  at the correct path
+// DOCKER_VOLUME_NAME="go"
+// docker volume create "$DOCKER_VOLUME_NAME"
+// drone exec --trusted --pipeline teleport-container-images-current-version-cron --clone=false --volume "${DOCKER_VOLUME_NAME}:/go"
+// # Cleanup
+// docker volume rm "$DOCKER_VOLUME_NAME"
+
+import (
+	"fmt"
+	"math"
+	"path"
+	"strings"
+)
+
+// If you are working on a PR/testing changes to this file you should configure the following for Drone testing:
+// 1. Publish the branch you're working on
+// 2. Set `PRBranch` to the name of the branch in (1)
+// 3. Set `ConfigureForPRTestingOnly` to true
+// 4. Create a public and private ECR, Quay repos for "teleport", "teleport-ent", "teleport-operator", "teleport-lab"
+// 5. Set `TestingQuayRegistryOrg` and `TestingECRRegistryOrg` to the org name(s) used in (4)
+// 6. Set the `ECRTestingDomain` to the domain used for the private ECR repos
+// 7. Create two separate IAM users, each with full access to either the public ECR repo OR the private ECR repo
+// 8. Create a Quay "robot account" with write permissions for the created Quay repos
+// 9. Set the Drone secrets for the secret names listed in "GetContainerRepos" to the credentials in (7, 8), prefixed by the value of `TestingSecretPrefix`
+//
+// On each commit, after running `make dronegen``, run the following commands and resign the file:
+// # Pull the current branch instead of v10 so the appropriate dockerfile gets loaded
+// sed -i '' "s~git checkout -qf \"\$(cat '/go/vars/full-version/v10')\"~git checkout -qf \"${DRONE_SOURCE_BRANCH}\"~" .drone.yml
+//
+// When finishing up your PR check the following:
+// * The testing secrets added to Drone have been removed
+// * `ConfigureForPRTestingOnly` has been set to false, and `make dronegen` has been reran afterwords
+
+const (
+	ConfigureForPRTestingOnly bool   = true
+	TestingSecretPrefix       string = "TEST_"
+	TestingQuayRegistryOrg    string = "fred_heinecke"
+	TestingECRRegistryOrg     string = "u8j2q1d9"
+	TestingEcrRegion          string = "us-east-2"
+	PRBranch                  string = "fred/arm-container-images"
+	ECRTestingDomain          string = "278576220453.dkr.ecr.us-east-2.amazonaws.com"
+)
+
+const (
+	ProductionRegistryOrg string = "gravitational"
+	PublicEcrRegion       string = "us-east-1"
+	StagingEcrRegion      string = "us-west-2"
+
+	LocalRegistry string = "drone-docker-registry:5000"
+)
+
+func buildContainerImagePipelines() []pipeline {
+	// These need to be updated on each major release.
+	latestMajorVersions := []string{"v10", "v9", "v8"}
+	branchMajorVersion := "v10"
+
+	if len(latestMajorVersions) == 0 {
```

Comment:
Seems unnecessary given the above code.

Reply:
it _probably_ is unnecessary. It caught a couple of dumb mistakes when I was building and debugging the pipelines but we probably don't need it outside of that.